packages/drivers/odsp-driver/src/odspDocumentDeltaConnection.ts

```diff
@@ -358,5 +358,10 @@ export class OdspDocumentDeltaConnection extends DocumentDeltaConnection impleme
             // If it's not critical error, we want to raise event on this object only.
             this.emit("disconnect", reason);
         }
+
+        // We set the closed flag as a part of the contract for overriding the disconnect method.  This is used by
+        // DocumentDeltaConnection to determine if emitting on the socket is allowed, which is important since
+        // OdspDocumentDeltaConnection reuses the socket rather than truly disconnecting it.
+        this.closed = true;
```

Comment:
It would be great to move it up. And unconditional to having socketReferenceKey.
Moving up ensures that if we have any kind of reentrancy (let's say someone sends an op in response to "disconnect" event), it will not get through.

Reply:
Moved up in latest.